packages/back-end/src/models/AgreementModel.ts

```diff
@@ -0,0 +1,59 @@
+import { MakeModelClass } from "back-end/src/models/BaseModel";
+import {
+  AgreementType,
+  agreementValidator,
+} from "back-end/src/validators/agreements";
+
+const BaseClass = MakeModelClass({
+  schema: agreementValidator,
+  collectionName: "agreements",
+  idPrefix: "agree_",
+  auditLog: {
+    entity: "agreement",
+    createEvent: "agreement.create",
+    updateEvent: "agreement.update",
+    deleteEvent: "agreement.delete",
+  },
+  globallyUniqueIds: false,
+  additionalIndexes: [
+    {
+      fields: {
+        organization: 1,
+        agreement: 1,
+      },
+      unique: false,
+    },
+  ],
+});
+
+export class AgreementModel extends BaseClass {
+  protected canRead(): boolean {
+    // TODO: should this be something else?  Perhaps readonly users shouldn't be able to access
+    // it as they shouldn't be able to call an endpoint that would need AI's help.
+    return true;
+  }
+  protected canCreate(): boolean {
+    return this.context.permissions.canManageOrgSettings();
+  }
+  protected canUpdate(): boolean {
+    return this.canCreate();
+  }
+  protected canDelete(): boolean {
+    return this.canCreate();
+  }
+
+  public getAgreementsForOrg = async () => {
+    const existing = await this._find({
+      organization: this.context.org.id,
+    });
+
+    return existing ? existing : [];
+  };
+  public getAgreementForOrg = async (agreement: AgreementType) => {
+    const existing = await this._findOne({
+      organization: this.context.org.id,
```

Comment:
You don't need to pass in `organization` here.  That is added automatically to every query.